src/operations/blas3/gemm_local.hpp

```diff
@@ -390,34 +424,56 @@ class Gemm<input_t, output_t, DoubleBuffer, NbcA, NbcB, ClSize, TileType,
         extract_input_blocks<check_m_limit, check_n_limit, false>(
             item_id, m, n, k, A, lda, B, ldb, s1, s3, out_of_range);
         id.barrier(cl::sycl::access::fence_space::local_space);
-        compute_block_gemm(s2, s4, reg_a, reg_b, reg_res);
-        A = A + cl_elems * (trans_a ? 1 : lda);
-        B = B + cl_elems * (trans_b ? ldb : 1);
-        k -= cl_elems;
+        compute_block_gemm<check_m_limit, check_n_limit>(item_id, s2, s4, reg_a,
+                                                         reg_b, reg_res);
+        A += cl_elems * (trans_a ? 1 : lda);
+        B += cl_elems * (trans_b ? ldb : 1);
+
         sync_smem<double_buffer, block_cols * ldsb, block_cols * ldsb,
                   ldsa * cl_elems, ldsa * cl_elems>(id, ofs, s1, s2, s3, s4);
+        k -= cl_elems;
       }
 
       if (k > 0) {
         extract_input_blocks<check_m_limit, check_n_limit, true>(
             item_id, m, n, k, A, lda, B, ldb, s1, s3, out_of_range);
         id.barrier(cl::sycl::access::fence_space::local_space);
-        compute_block_gemm(s2, s4, reg_a, reg_b, reg_res);
+        compute_block_gemm<check_m_limit, check_n_limit>(item_id, s2, s4, reg_a,
+                                                         reg_b, reg_res);
+
         sync_smem<double_buffer, block_cols * ldsb, block_cols * ldsb,
                   ldsa * cl_elems, ldsa * cl_elems>(id, ofs, s1, s2, s3, s4);
       }
 
       // store the output
-      store_output_block<check_m_limit, check_n_limit>(mc, nc, C, ldc, reg_res,
-                                                       out_of_range);
+      store_output_block<check_m_limit, check_n_limit>(item_id, mc, nc, C, ldc,
+                                                       reg_res, out_of_range);
       orig_A += (a_size * batch_stride);
       orig_B += (b_size * batch_stride);
       orig_C += (c_size * batch_stride);
       // batch_size_ must be signed as the negative value has meaning here.
-      batch_size_ -= batch_stride;
-    } while (batch_size_ > wg_batch_id);
+      batch_size -= batch_stride;
+    } while (batch_size > wg_batch_id);
   }
 
+  template <bool internal, index_t p_size = packetize_t::packet_size,
+            typename OutputPointerType>
+  SYCL_BLAS_INLINE typename std::enable_if<!internal>::type store_packet(
+      element_t *reg, OutputPointerType out_ptr) {
+    *out_ptr = alpha_ * *reg;
```

Comment:
This is correct, however,  for readability, it would be nice to wrap *rage in parenthesis. *reg -> (*reg);

Reply:
Good point, done!